### DISCUSSION TITLE
Use different input parameter based on a condition

### DISCUSSION THREAD

**Discussion by dingo-d** [community/community#26557] [Actions]
<p>I am trying to run deploy action for my staging and production, and I’d like to use different SSH keys which are added as GH secret, based on which branch I’m on.</p>
<p>So I added a ‘stage check’ step before my other steps in the deployment process that looks like this</p>
<pre><code class="lang-auto">    steps:
      - name: Stage check
        id: current_stage
        run: |
          if [ ${{ github.ref == 'refs/heads/main' }} ]; then
            echo "::set-output name=STAGE::production"
            echo "::set-output name=SSH_KEY::${{ secrets.PRODUCTION_KEY }}"
          elif [ ${{ github.ref == 'refs/heads/staging' }} ]; then
            echo "::set-output name=STAGE::staging"
            echo "::set-output name=SSH_KEY::${{ secrets.STAGING_KEY }}"
          fi
</code></pre>
<p>In my other steps, I can now get access to</p>
<pre><code class="lang-auto">${{ steps.current_stage.outputs.SSH_KEY }}
</code></pre>
<p>and</p>
<pre><code class="lang-auto">${{ steps.current_stage.outputs.STAGE }}
</code></pre>
<p>I now have</p>
<pre><code class="lang-auto">      - name: Add SSH key
        uses: webfactory/ssh-agent@v0.5.2
        with:
            ssh-private-key: ${{ steps.current_stage.outputs.SSH_KEY }}
</code></pre>
<p>But this seems to fail. Not 100% sure why, but my hunch is that the output cannot see the encrypted secret content, so when it’s being used in the parameter <code>ssh-private-key</code>, it’s not passed as a decrypted value and then I get a failure.</p>
<p>So what I’m wondering if it’s possible to do, in one step, to have something like</p>
<pre><code class="lang-auto">      - name: Add SSH key
        uses: webfactory/ssh-agent@v0.5.2
        if: ${{ github.ref == 'refs/heads/main' }}
        with:
            ssh-private-key: ${{ secrets.PRODUCTION_KEY }}
        if: ${{ github.ref == 'refs/heads/staging' }}
        with:
            ssh-private-key: ${{ secrets.STAGING_KEY }}
</code></pre>
<p>So that my input argument to the <code>webfactory/ssh-agent</code> action is different based on that condition?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I see two issues here:</p>
<ol>
<li>The conditionals in your <code>current_stage</code> step are mixing Bash and GitHub Actions syntax, and the result of that is probably not what you want (both are always true, so you always get the production path because it’s checked first). You should use Actions syntax (the <code>${{ ... }}</code> part) only to substitute the <code>github.ref</code> and Bash for the rest:</li>
</ol>
<pre><code class="lang-yaml">        run: |
          if [ ${{ github.ref }} = "refs/heads/main" ]; then
            echo "::set-output name=STAGE::production"
            echo "::set-output name=SSH_KEY::${{ secrets.PRODUCTION_KEY }}"
          elif [ ${{ github.ref }} = "refs/heads/staging" ]; then
            echo "::set-output name=STAGE::staging"
            echo "::set-output name=SSH_KEY::${{ secrets.STAGING_KEY }}"
          fi
</code></pre>
<ol start="2">
<li>The OpenSSH private keys I know are multi-line files. Running that through the output without URL encoding (so there are no newlines) will mess them up. The easiest solution is probably to replace all newlines with <code>%0A</code> in the secrets.</li>
</ol>

**Comment by dingo-d** [comment]
<p>Yeah, I got the production stage all the time. I’ve fixed this, but the key one was tricky, so I separated this to 4 steps, all of them have the <code>if: ${{ github.ref == 'refs/heads/staging' }}</code> checks which work.</p>
<p>Thanks for the help!</p>
